programs/sig-aggregator/src/error.rs

```diff
@@ -2,9 +2,276 @@ use thiserror::Error;
 
 #[derive(Error, Debug)]
 pub enum AggregatorError {
-    #[error("Configuration error: {0}")]
-    Config(String),
+    /// Configuration-related errors
+    #[error("Configuration error: {message}")]
+    Config {
```

Comment:
Seems like this should just be `InvalidConfig` which is our own direct checks, and if there are 3rd party libraries that also causes any specific errors (that are not directly related to just checking the config for validity), then those should be handled differently (i.e. remove the source here - I think)